clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -607,6 +608,14 @@ public synchronized void handleCompletedBatch(ProducerBatch batch, ProduceRespon
         removeInFlightBatch(batch);
     }
 
+    public synchronized void transitionToUninitialized(RuntimeException exception) {
+        transitionTo(State.UNINITIALIZED);
+        lastError = null;
+        if (pendingTransition != null) {
```

Comment:
Do you have more background on what pendingTransitions we typically have? I don't think this hurts unless it will cause a fatal error. If it does, we may need to take a closer look.

Reply:
thanks, i think it's fine to leave it here.

Reply:
I'd prefer you confirm that it's fine. Looking at the code -- it seems to set the TransactionalRequestResult state to the error and count down the latch. Given that it is for the request -- looks to be fine.